### PR TITLE
feat: add rule for upper-snake case enum cases

### PR DESCRIPTION
**Pull request by owenvoke**

### CODE REVIEW

File: rfcs/20230210-upper-snake-case-enum-cases.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Upper-snake case enum cases
```
Comment (olivernybroe) [inline rfcs/20230210-upper-snake-case-enum-cases.md:1]: Isn't the casing normally just called `UPPER_CASE`? 🤔

**Review by olivernybroe — APPROVED** [review]
RFC looks good 👍 

Could you add something into the `rules.md` file also with the change?

File: rfcs/20230210-upper-snake-case-enum-cases.md (hunk 1 above)
Comment (owenvoke) [reply rfcs/20230210-upper-snake-case-enum-cases.md:1]: I couldn't find a good reference to this. 🤷🏻 I assumed upper-case would mean `UPPER CASE` 🤔

Comment (olivernybroe) [reply rfcs/20230210-upper-snake-case-enum-cases.md:1]: Ah right yeah, let's keep the wording `UPPER_SNAKE`, but write it like that :) 

I often prefer writing the casing names in the casing they are, to make it very clear 👍

File: rules.md [hunk 2]
```diff
@@ -213,3 +213,27 @@ type userInvoice {
 type UserInvoice {
 }
 ```
+
+## `UPPER_SNAKE` enum cases ![linting](https://img.shields.io/badge/linting-auto-blue)
+
+All enum cases **MUST** use upper case.
```
Comment (olivernybroe) [inline rules.md:219]:
```suggestion
All enum cases **MUST** use upper snake case.
```

**Review by olivernybroe — APPROVED** [review]

**Pull request merged**